hs_app_timeseries/models.py

```diff
@@ -128,52 +367,83 @@ def metadata(self):
         # phase-1 of implementation supports only sqlite file
         return (".sqlite",)
 
-
     @classmethod
     def can_have_multiple_files(cls):
         # can have only 1 file
         return False
 
 
-# this would allow us to pick up additional form elements for the template before the template is displayed
+# this would allow us to pick up additional form elements for the template before the template
+# is displayed
 processor_for(TimeSeriesResource)(resource_processor)
 
 
 class TimeSeriesMetaData(CoreMetaData):
-    _site = GenericRelation(Site)
-    _variable = GenericRelation(Variable)
-    _method = GenericRelation(Method)
-    _processing_level = GenericRelation(ProcessingLevel)
-    _time_series_result = GenericRelation(TimeSeriesResult)
+    _sites = GenericRelation(Site)
+    _variables = GenericRelation(Variable)
+    _methods = GenericRelation(Method)
+    _processing_levels = GenericRelation(ProcessingLevel)
+    _time_series_results = GenericRelation(TimeSeriesResult)
 
     @property
-    def site(self):
-        return self._site.all().first()
+    def is_dirty(self):
+        # this property is used to determine if the SQLite file needs
+        # to be synced with metadata in database
+        if self.resource.files.first() is None:
+            # the SQLite file is missing - so nothing to update
+            return False
+        dirty = any(site.is_dirty for site in self.sites) or any(
+            variable.is_dirty for variable in self.variables) or any(
+            method.is_dirty for method in self.methods) or any(
+            processing_level.is_dirty for processing_level in self.processing_levels) or any(
+            ts_result.is_dirty for ts_result in self.time_series_results) or any(
+            cv_variable_name.is_dirty for cv_variable_name in
+            CVVariableName.objects.filter(metadata=self)) or any(
+            cv_variable_type.is_dirty for cv_variable_type in
+            CVVariableType.objects.filter(metadata=self)) or any(
+            cv_speciation.is_dirty for cv_speciation in
+            CVSpeciation.objects.filter(metadata=self)) or any(
+            cv_site_type.is_dirty for cv_site_type in
+            CVSiteType.objects.filter(metadata=self)) or any(
+            cv_elev_datum.is_dirty for cv_elev_datum in
+            CVElevationDatum.objects.filter(metadata=self)) or any(
+            cv_method_type.is_dirty for cv_method_type in
+            CVMethodType.objects.filter(metadata=self)) or any(
+            cv_units_type.is_dirty for cv_units_type in
+            CVUnitsType.objects.filter(metadata=self)) or any(
+            cv_status.is_dirty for cv_status in CVStatus.objects.filter(metadata=self)) or any(
+            cv_medium.is_dirty for cv_medium in CVMedium.objects.filter(metadata=self)) or any(
+            cv_agg_stat.is_dirty for cv_agg_stat in
+            CVAggregationStatistic.objects.filter(metadata=self))
+
+        return dirty
```

Comment:
@pkdash Is there a way to set is_dirty flag in this ```TimeSeriesMetaData``` place directly from each individual metadata element? It looks cumbersome to check each individual metadata element is_dirty flag to determine whether the resource metadata is dirty here. I think you can define is_dirty flag in one place and have each individual element to set this single flag to avoid this cumbersome check here. If you need to know which specific metadata element is dirty, you can define a dirty element list to collect all dirty metadata elements. I think there should be a better way to achieve this.